File: c_sharp/MarsRover/MarsRover/Grid.cs
using System.Collections;

namespace MarsRover
{
    public class Grid
    {
        public Position move(Position position, Direction direction)
        {
            switch(direction)
            {
                case Direction.N:
                    return new Position(position.X, position.Y + 1);
                case Direction.E:
                    return new Position(position.X + 1, position.Y);
                case Direction.S:
                    return new Position(position.X, position.Y - 1);
                default:
                    return new Position(position.X - 1, position.Y);
            }
        }
    }
}

File: python/MarsRover/src/Grid.py
from src.Direction import Direction
from src.Position import Position


class Grid:
    def __init__(self):
        pass

    def move(self, direction, position):
        switcher = {
            Direction.NORTH: Position(position.x, position.y + 1),
            Direction.EAST: Position(position.x - 1, position.y),
            Direction.SOUTH: Position(position.x, position.y - 1),
            Direction.WEST: Position(position.x + 1, position.y - 1)
        }

        return switcher.get(direction, position)